x/crosschain/keeper/evm_deposit.go

```diff
@@ -24,7 +24,10 @@ const InCCTXIndexKey = "inCctxIndex"
 // returns (isContractReverted, err)
 // (true, non-nil) means CallEVM() reverted
 func (k Keeper) HandleEVMDeposit(ctx sdk.Context, cctx *types.CrossChainTx) (bool, error) {
-	to := ethcommon.HexToAddress(cctx.GetCurrentOutboundParam().Receiver)
+	to, err := cctx.GetValidReceiverAddress()
```

Comment:
ctx.GetCurrentOutboundParam().Receiver will always contains the receiver moving forward, and since it is on ZetaChain, the address should always be EVM format

I think we could implement a function such as https://github.com/zeta-chain/node/issues/2789 and use it here with receiver address

Reply:
THe function is a good idea .
This is really the only change in logic that the PR implements, IMO if we want to change this , it would make sense to discard this pr for now , and revisit this once we have everything related to v2 functional